main.tf

```diff
@@ -13,16 +16,16 @@ locals {
 # Security group with name
 ##########################
 resource "aws_security_group" "this" {
-  count = var.create && false == var.use_name_prefix ? 1 : 0
+  count = var.create && var.create_group && !var.use_name_prefix ? 1 : 0
 
-  name                   = var.name
-  description            = var.description
+  name                   = local.name
```

Comment:
These 2 new locals are not necessary. You can revert this change.

Reply:
Done